mobile/geth.go

```diff
@@ -179,6 +179,20 @@ func NewNode(datadir string, config *NodeConfig) (stack *Node, _ error) {
 				config.EthereumNetworkID = 5
 			}
 		}
+		// If we have the Bor Mainnet, hard code the chain configs too
+		if config.EthereumGenesis == BorMainnetGenesis() {
+			genesis.Config = params.BorMainnetChainConfig
+			if config.EthereumNetworkID == 1 {
+				config.EthereumNetworkID = 137
+			}
+		}
+		// If we have the Mumbai testnet, hard code the chain configs too
+		if config.EthereumGenesis == MumbaiGenesis() {
+			genesis.Config = params.MumbaiChainConfig
+			if config.EthereumNetworkID == 1 {
+				config.EthereumNetworkID = 80001
+			}
+		}
```

Comment:
Can you explain more why this condition is required?

Reply:
As per the code and comments in above file, I can see that this comes into the effect when the
```
	if config.EthereumGenesis != "" {
```
and it parses the user-supplied genesis spec, Based on that it check what network hash it matches and update the config accordingly.